internal/application/application.go

```diff
@@ -85,19 +85,23 @@ func Exists(ctx context.Context, cluster *kubernetes.Cluster, app models.AppRef)
 	return true, nil
 }
 
-// CurrentlyStaging returns true if there is an active (not completed) Job
-// for this application.
+// CurrentlyStaging returns true if there is an active Job for this application.
 func CurrentlyStaging(ctx context.Context, cluster *kubernetes.Cluster, namespace, appName string) (bool, error) {
 
-	l, err := cluster.ListJobs(ctx, helmchart.StagingNamespace, fmt.Sprintf("app.kubernetes.io/name=%s,app.kubernetes.io/part-of=%s", appName, namespace))
+	selector := fmt.Sprintf("app.kubernetes.io/name=%s,app.kubernetes.io/part-of=%s",
+		appName, namespace)
 
+	jobList, err := cluster.ListJobs(ctx, helmchart.StagingNamespace, selector)
 	if err != nil {
 		return false, err
 	}
 
-	// assume that completed jobs are from the past and have a CompletionTime
-	for _, pr := range l.Items {
-		if pr.Status.CompletionTime == nil {
+	for _, job := range jobList.Items {
+		done, err := cluster.IsJobDone(ctx, job.Name, helmchart.StagingNamespace)
```

Comment:
This is `N+1` requests (one to fetch all jobs and 1 request per Job to check if it's done). I believe the Jobs in `jobList.Items` already hold the information needed in order to tell if they are done or not. This can be solved if `IsJobDone` takes a Job as an argument instead of a string. This means, the caller will have to fetch the Job.

Reply:
We delete old Jobs so it should be a maximum of 3 requests but I think it still makes sense to fix.

Reply:
Checking. Note that we should have `N == 1`.

Reply:
Easier to inline, also, can undo the complexification in the helper code.

Reply:
Done. Commit 8b803e28fde413bad1103bc10b813f448b7484a9